src/test/java/com/werfen/report/FormReportTest.java

```diff
@@ -32,7 +33,19 @@ public void generateFormReport() throws IOException, ReportException {
     }
 
     @Test
-    public void generateFormReportWithLessFields() throws IOException, ReportException {
+    public void generateStreamFormReport() throws IOException, ReportException {
+        String fileName = "form_report";
+
+        FormReportService formReportService = new FormReportService();
+        ByteArrayOutputStream report = formReportService.buildToStream(this.getConfiguration(null), this.getData(), PageFormat.A4);
+
+        InputStream actualStream = new ByteArrayInputStream(report.toByteArray());
+        InputStream expectedStream = Files.newInputStream(Paths.get(GOLDEN_PATH + fileName + GOLDEN_SUFFIX + ReportFormat.PDF.getFileExtension()));
+        ComparisonResultAssertions.assertEquals(PDFComparator.compareFiles(expectedStream, actualStream));
```

Comment:
Shouldn't the comparator method be better named compareStreams?

Reply:
It is really comparing files, but if you look at the jasper-report-test-utilities library implementation, the compareStreams method is available and is called by the compareFiles. I used that because we are using reference files for compaison so it's just a matter of convenience.